docs/proposals/api-grouping.md

```diff
@@ -0,0 +1,375 @@
+  - PaaSes on top of Kubernetes might to do this.
+- So that users can focus on learning one group at a time.
+- So that security/policy admins can audit/enable an API group at a time, and not need to
+  review all kinds at once.
+- So that different groups within the kubernetes project can have a greater degree of autonomy
+  - Conways law 
+  - have trusted api reviewers per group.
+
+## Ways to Group things
+
+Listed below are some ways that you might want to group things.
+It is not a proposal to do all those things; in fact some are in conflict.
+
+### Proposal from #635
+
+Issue #635 proposed this grouping:
+
+```
+apis/podScheduling/v1
+	Pods
+	PodTemplates
+	Bindings
+
+apis/services/v1
+	Services
+
+apis/rcs/v1
+	ReplicationController
+
+apis/nodes/v1
+	Nodes
+```
+
+### Admin vs User
+
+Issue #3806 proposed splitting "Cluster" and
+"User" resources.  Applied to the current state,
+that would look something like this:
+
+```
+/api/v1
+	persistentVolumeClaim
+	pod
+	podTemplate
+	replicationController
+	service
+	serviceAccount
+	endpoints
+	event
+	secret
+	petSet
+	workflow
+	scheduledJob
+	componentStatus   # Maybe goes in apis/cluster?
+	ingress           # Maybe goes in apis/cluster?
+	podSecurityPolicyBinding
+
+/apis/metrics/v1alpha1
+	Pod
+
+/apis/cluster/v1
+	node
+	namespace
+	binding
+	persistentVolume
+	limitRange
+	resourceQuota
+	daemonSet
+	clusterAutoscaler
+	metadataPolicy
+	podSecurityPolicy
+
+/apis/clustermetrics/v1alpha1
+	Pod
+	Node
+
+/apis/extensions/v1beta1
+	scale
+	horizontalPodAutoscaler
+	thirdPartyResource
+	thirdPartyResourceData
+	deployment
+	job
+	replicaSet
+	scaleStatus
+	scale
+```
+
+Evaluation:
+
+- Everything still in the main API (except componentStatus and maybe podSecurityPolicy)
+  is stuff that makes sense to namespace, since they are things you replicate if
+  you make a "copy" of a service or system.
+- Everything in the `apis/cluster/v1` group does not need a namespace,
+  except clusterAutoscaler, limitRange, and resourceQuota
+- Everything in `apis/cluster/v1` is ostensibly stuff that a tenant does not need
+  to view or modify to use the system.  And it is stuff that the a hosted
+  kubernetes provider certainly wants tight control over, and may want
+  to hide so that it can be implemented differently.
+- non-admins can focus their learning on the User group.
+
+### Vertical Grouping
+
+Group things that work with the same type of workloads.
+
+Sketch:
+```
+/apis/batch/v1
+	scheduledJob
+	workflow
+	job
+
+/apis/longrunning/v1  # need better name
+	replicaSet
+	deployment
+	petSet
+
+/apis/cluster/v1
+	daemonSet
+
+# ReplicationController is deprecated by ReplicaSet.
+```
+
+Rationale for `batch` group is that if you are running a batch job, you are likely to care about
+all three things (run-to-completion, start-at-a-time, and dependencies).
+
+Rationale for `DaemonSet` being in separate group is that typically only the cluster admin
```

Comment:
It is problematic to allow arbitrary users to create DaemonSets, except on dedicated machines, and the meaning of DaemonSet is inherently tied to Nodes, so it does make sense to separate it from the more purely user-oriented, application-oriented APIs.